src/Security/SimpleLoginAuthenticator.php

```diff
@@ -0,0 +1,178 @@
+        private readonly ImageManager $imageManager,
+        private readonly ImageFactory $imageFactory,
+        private readonly ImageRepository $imageRepository,
+        private readonly IpResolver $ipResolver,
+        private readonly Slugger $slugger,
+        private readonly UserRepository $userRepository,
+        private readonly SettingsManager $settingsManager
+    ) {
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return 'oauth_simplelogin_verify' === $request->attributes->get('_route');
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        $client = $this->clientRegistry->getClient('simplelogin');
+        $slugger = $this->slugger;
+
+        $provider = $client->getOAuth2Provider();
+
+        $accessToken = $provider->getAccessToken('authorization_code', [
+            'code' => $request->query->get('code'),
+        ]);
+
+        $rememberBadge = new RememberMeBadge();
+        $rememberBadge = $rememberBadge->enable();
+
+        return new SelfValidatingPassport(
+            new UserBadge($accessToken->getToken(), function () use ($accessToken, $client, $slugger) {
+                /** @var SimpleLoginResourceOwner $simpleloginUser */
+                $simpleloginUser = $client->fetchUserFromToken($accessToken);
+
+                $existingUser = $this->entityManager->getRepository(User::class)->findOneBy(
+                    ['oauthSimpleLoginId' => $simpleloginUser->getId()]
+                );
+
+                if ($existingUser) {
+                    return $existingUser;
+                }
+
+                $user = $this->userRepository->findOneBy(['email' => $simpleloginUser->getEmail()]);
+
+                if ($user) {
+                    $user->oauthSimpleLoginId = $simpleloginUser->getId();
+
+                    $this->entityManager->persist($user);
+                    $this->entityManager->flush();
+
+                    return $user;
+                }
+
+                if (false === $this->settingsManager->get('MBIN_SSO_REGISTRATIONS_ENABLED')) {
+                    throw new CustomUserMessageAuthenticationException('MBIN_SSO_REGISTRATIONS_ENABLED');
+                }
+
+                $email = $simpleloginUser->toArray()['email'];
+                $username = $slugger->slug(substr($email, 0, strrpos($email, '@')));
+
+                if ($this->userRepository->count(['username' => $username]) > 0) {
+                    $username .= rand(1, 999);
+                }
+
+                $dto = (new UserDto())->create(
+                    $username,
+                    $simpleloginUser->getEmail()
+                );
+
+                $avatar = $this->getAvatar($simpleloginUser->getPictureUrl());
+
+                if ($avatar) {
+                    $dto->avatar = $this->imageFactory->createDto($avatar);
+                }
+
+                $dto->plainPassword = bin2hex(random_bytes(20));
```

Comment:
what is this?

Reply:
Which line, the password? All our SSO options do this... because the user needs a password (albeit random) so the field isn't null:

![image](https://github.com/MbinOrg/mbin/assets/35878315/93b8d660-3eec-49a1-ae29-5030d8878556)

The API does as well, but it's a little "tougher":

![image](https://github.com/MbinOrg/mbin/assets/35878315/40d39833-d385-43c7-a16e-e854c8624cea)

I guess we could argue if the way it's done now has enough entropy so as to not be brute forced...

Reply:
yea indeed the password, but I understand now. Thanks!